---
title: "Depression_Microbiome_Data"
author: "Bahia Chahwan"
date: "22/08/2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r load packages}
library("phyloseq")
library("ggplot2")
library("vegan")
library("reshape2")
library("tidyverse")
library("grid")
library("gridExtra")
library("zoo")
```


#Setting up in R
```{r Setting up in R}

#otu clustered and chimeras removed
biom<-"data/otu_table_final1_json.biom"
map<-read.table("data/mapping_depression_newR.txt", sep="\t", header=TRUE, row.names=1)
#had to run this comand as the other one (^) wouldnt work for some reason 
#map<-read.table(file.choose("data/mapping_depression_newR.txt"), sep="\t", header=TRUE, row.names=1)

tree<-read_tree("data/rep_set.tre")

## Import Data to PhyloSeq
depression2017<-import_biom(biom, treefilename=tree)

depression_map<-sample_data(map)

depression2017<-merge_phyloseq(depression2017,depression_map)

#18696 taxa and 142 samples
depression2017
head(depression2017)
colnames(tax_table(depression2017)) = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species")

```


#Cleaning up data
```{r Cleaning up data}

#Removed participant incorrectly enrolled and mock 
depression2017_1<-subset_samples(depression2017, !(SampleID.1 %in% c("P9.pre","P7.pre","P7.post", "mock")))

#18695 taxa and 138 samples
depression2017_1

#Rarefy to 28170 - lose p4.pre
depression2017_rare<-rarefy_even_depth(depression2017_1, sample.size=28170, rngseed=711, replace=FALSE, trimOTUs=TRUE)
##7 samples removed because they contained fewer reads than `sample.size`
##2364OTUs were removed because they are no longer present in any sample after random subsampling

#16331 taxa and 131 samples 
depression2017_rare

#filter out rare OTUs
depression2017_fil<- filter_taxa(depression2017_rare, function(x) sum(x)>=10 & sum(x>0)>0.25*length(x), TRUE)

#938 taxa and 131 samples
depression2017_fil
```


#Rarefaction Curve
```{r Rarefaction Curve}

#Create a rarefaction curve to find the appropraite place for rarefying your data. Uses vegan for calculating the curve, and base R for plotting. 
#Plot rarecurve
t_otu_depression2017<-as.matrix(t(otu_table(depression2017_1)))
curve<-rarecurve(t_otu_depression2017, step=500, label=FALSE, xlim=c(0,50000), ylim=c(0,2000), ylab = "OTU Count", xaxt="n")

label(title="Rarefaction Curve of All Microbiome Samples")

axis(1)
#place a line over the curve where you proppose to rarefy
abline(v=28170, col="red")

```


#subsetting phyloseq objects
##havn't done this yet
```{r, subsetting phyloseq objects}

#You many want to remove certain samples or OTUs from the analysis.  There are several ways to do this in Phyloseq, all require creating a new Phyloseq object.
#First example, subset to samples that match a string in a column of the sample data (i.e. the mapping file)
#depression2017_rare_controls<-subset_samples(depression2017_rare, Participant=="con")
#sample_data(practice2017_rare_controls)
#Second example, subset to samples that match everything except the string in a column of the sample data
#practice2017_rare_disease<-subset_samples(practice2017_rare, Disease_status!="control")
#Now check the sample data for your new object to make sure it did what you think it did.  In this example I'm only returning the column that lists Disease_status
#sample_data(practice2017_rare_disease)$Disease_status
#Third example, subset to samples that match several strings in a column of the sample data
#practice2017_rare_disease2<-subset_samples(practice2017_rare, #Disease_status%in%c("mdd", "recovering_mdd"))
#And check the sample data as before.
#sample_data(practice2017_rare_disease2)$Disease_status
#Forth example, subset to samples that match everything except several strings in a column of the sample data
#practice2017_rare_controls2<-subset_samples(practice2017_rare, !(Disease_status%in%c("mdd", "recovering_mdd")))
#And check the sample data as before.
#sample_data(practice2017_rare_controls2)$Disease_status
```

#Alpha diversity analysis
```{r, alpha diversity}
#participant (pre, post, con)

#Once you have rarefied your data, explore the level of alpha diversity.  Uses phyloseq plot function which is built on ggplot2.  The x value will determine how samples are grouped along the x axis.
plot_richness(depression2017_fil, x="Participant", measures=c("Observed", "Chao1","Shannon"))+ geom_boxplot()+xlab("Group")+ylab("Diversity")

dep_con_richness<-subset_samples(depression2017_fil, Dep_Con %in% c("Dep", "Con"))


#alpha diversity richness pre and con graph 
plot_richness(dep_con_richness, x="Dep_Con", measures=c("Observed", "Chao1","Shannon"))+ geom_boxplot()+xlab("Group")+ylab("Diversity")+labs(title="Diversity of Baseline Depressed Samples and Healthy Controls")+scale_x_discrete(limits=c("Dep", "Con"))


allocation_groups_richness<-subset_samples(depression2017_fil, Allocation_Groups %in% c("PreA", "PreB", "PostA", "PostB"))
plot_richness(allocation_groups_richness, x="Allocation_Groups", measures=c("Observed", "Chao1","Shannon"))+ geom_boxplot()+xlab("Group")+ylab("Diversity")

##two graphs below richness for pre and post samples allocation a and b
allocation_a_richness<-subset_samples(depression2017_fil, Allocation_Groups %in% c("PreA", "PostA"))
plot_richness(allocation_a_richness, x="Allocation_Groups", measures=c("Observed", "Chao1","Shannon"))+ geom_boxplot()+xlab("Group")+ylab("Diversity")+scale_x_discrete(limits=c("PreA", "PostA"))+labs(title="Pre and Post Depressed Particpant Samples Treated with Probiotics")

allocation_b_richness<-subset_samples(depression2017_fil, Allocation_Groups %in% c("PreB", "PostB"))
plot_richness(allocation_b_richness, x="Allocation_Groups", measures=c("Observed", "Chao1","Shannon"))+ geom_boxplot()+xlab("Group")+ylab("Diversity")+scale_x_discrete(limits=c("PreB", "PostB"))+labs(title="Pre and Post Depressed Particpant Samples Treated with Placebo")

#Convert alpha diversity estimates to a data form we can use for statistical testing
depression_rare_richness<-as.matrix(estimate_richness(depression2017_fil, measures=c("Observed","Chao1","Shannon")))

##don't need this - row names are correct
#row.names(practice2017_rare_richness)<-row.names(sample_data(practice2017_rare))

depression_rare_richness<-cbind(depression_rare_richness,sample_data(depression2017_fil))

depression_rare_richness

#For comaparisons of alpha diversity of more than two groups, use the Kruskal-Wallis test 
#Test if alpha diversity (observed) is significantly different between Disease_status groups
kruskal.test(Observed ~ Participant, data=depression_rare_richness)
##Kruskal-wallis rank sum test, pvalue 0.75498, thus no significant difference between populations, null hypothesis assumes samples are from identical populations

kruskal.test(Chao1 ~ Participant, data=depression_rare_richness)

kruskal.test(Shannon ~ Participant, data=depression_rare_richness)

#Now test if diversity is significantly different between controls and patients who were ever depressed (and recovering mdd groups (Original_status)
#only two groups here so use Man-Whitney U test
#null asserts that medians of two samples are indentical 
wilcox.test(Observed ~ Dep_Con, data=depression_rare_richness)
##data:  Observed by Dep_Con W = 769.5, p-value = 0.3755
##alternative hypothesis: true location shift is not equal to 0

wilcox.test(Chao1 ~ Dep_Con, data=depression_rare_richness)
wilcox.test(Shannon ~ Dep_Con, data=depression_rare_richness)

wilcox.test(Observed ~ Con_A, data=depression_rare_richness)
##data:  Observed by Con_A W = 180, p-value = 0.4416


##Warning message:
##In wilcox.test.default(x = c(412, 401, 409, 445, 381, 434, 511,  :
  ##cannot compute exact p-value with ties

wilcox.test(Observed ~ Con_B, data=depression_rare_richness)
##data:  Observed by Con_B W = 216, p-value = 0.9297

wilcox.test(Observed ~ Pre_ProA, data=depression_rare_richness)
wilcox.test(Shannon ~ Pre_ProA, data=depression_rare_richness)
wilcox.test(Chao1 ~ Pre_ProA, data=depression_rare_richness)
##data:  Observed by Pre_ProA W = 239, p-value = 0.654

wilcox.test(Observed ~ Pre_PostB, data=depression_rare_richness)
wilcox.test(Chao1 ~ Pre_PostB, data=depression_rare_richness)
wilcox.test(Shannon ~ Pre_PostB, data=depression_rare_richness)
##data:  Observed by Pre_PostB W = 265.5, p-value = 0.5892

wilcox.test(Observed ~ ProA_B, data=depression_rare_richness)
##data:  Observed by ProA_B W = 203, p-value = 0.504

##PRE_BDI_LEVELS - severe, mild moderate
kruskal.test(Observed ~ PRE_BDI_LEVELS, data=depression_rare_richness)
##Kruskal-Wallis chi-squared = 3.1845, df = 2, p-value = 0.2035

kruskal.test(Observed ~ MINI, data=depression_rare_richness)
kruskal.test(Shannon ~ MINI, data=depression_rare_richness)
kruskal.test(Chao1 ~ MINI, data=depression_rare_richness)

```


#Beta Diversity
```{r Beta Diversity}

depression2017_wuf<-distance(depression2017_fil, "wunifrac", type="samples")
##Warning message:
##In UniFrac(physeq, weighted = TRUE, ...) :
  ##Randomly assigning root as -- 531539 -- in the phylogenetic tree in the data you provided.

depression2017_wuf_ord<-ordinate(depression2017_fil, method="PCoA", distance=depression2017_wuf)

plot_ordination(depression2017_fil, depression2017_wuf_ord, type="samples",color="Participant")

pre_con_rare_fil<-subset_samples(depression2017_fil, Participant %in% c("Pre","Con"))
pre_con_ord<-ordinate(pre_con_rare_fil, method="PCoA", "wunifrac")


#beta-diversity pre and con graph 
plot_ordination(pre_con_rare_fil,pre_con_ord, color="Participant") + stat_ellipse()+labs(title="Weighted Unifrac PCoA Baseline Depressed vs Controls")

dep_con_dist<-distance(pre_con_rare_fil, "wunifrac")

adonis(dep_con_dist ~ Dep_Con + Extraction_batch + PCR_batch, data=data.frame(sample_data(pre_con_rare_fil)))

con_A_rare_fil<-subset_samples(depression2017_fil, Con_A %in% c("Con","A"))
con_A_ord<-ordinate(con_A_rare_fil, method="PCoA", "wunifrac")

plot_ordination(con_A_rare_fil,con_A_ord, color="Con_A")

con_A_dist<-distance(con_A_rare_fil, "wunifrac")

adonis(con_A_dist ~ Con_A + Extraction_batch + PCR_batch, data=data.frame(sample_data(con_A_rare_fil)))


con_B_rare_fil<-subset_samples(depression2017_fil, Con_B %in% c("Con","B"))
con_B_ord<-ordinate(con_B_rare_fil, method="PCoA", "wunifrac")

plot_ordination(con_B_rare_fil,con_B_ord, color="Con_B")

con_B_dist<-distance(con_B_rare_fil, "wunifrac")

adonis(con_B_dist ~ Con_B + Extraction_batch + PCR_batch, data=data.frame(sample_data(con_B_rare_fil)))

pre_proA_rare_fil<-subset_samples(depression2017_fil, Pre_ProA %in% c("pre","A"))
pre_proA_ord<-ordinate(pre_proA_rare_fil, method="PCoA", "wunifrac")

plot_ordination(pre_proA_rare_fil, pre_proA_ord, color="Pre_ProA")+

pre_proA_dist<-distance(pre_proA_rare_fil, "wunifrac")

adonis(pre_proA_dist ~ Pre_ProA + Extraction_batch + PCR_batch, data=data.frame(sample_data(pre_proA_rare_fil)))

pre_proB_rare_fil<-subset_samples(depression2017_fil, Pre_PostB %in% c("pre","B"))
pre_proB_ord<-ordinate(pre_proB_rare_fil, method="PCoA", "wunifrac")

plot_ordination(pre_proB_rare_fil, pre_proB_ord, color="Pre_PostB")

pre_proB_dist<-distance(pre_proB_rare_fil, "wunifrac")

adonis(pre_proB_dist ~ Pre_PostB + Extraction_batch + PCR_batch, data=data.frame(sample_data(pre_proB_rare_fil)))

proA_B_rare_fil<-subset_samples(depression2017_fil, ProA_B %in% c("A","B"))
proA_B_ord<-ordinate(proA_B_rare_fil, method="PCoA", "wunifrac")

plot_ordination(proA_B_rare_fil, proA_B_ord, color="ProA_B")

proA_B_dist<-distance(proA_B_rare_fil, "wunifrac")

adonis(proA_B_dist ~ ProA_B + Extraction_batch + PCR_batch, data=data.frame(sample_data(proA_B_rare_fil)))

#severity vs moderate vs mild
pre_bdi_levels_rare_fil<-subset_samples(depression2017_fil, PRE_BDI_LEVELS %in% c("severe","moderate", "mild"))
pre_bdi_levels_ord<-ordinate(pre_bdi_levels_rare_fil, method="PCoA", "wunifrac")

plot_ordination(pre_bdi_levels_rare_fil, pre_bdi_levels_ord, color="PRE_BDI_LEVELS")

pre_bdi_dist<-distance(pre_bdi_levels_rare_fil, "wunifrac")

adonis(pre_bdi_dist ~ PRE_BDI_LEVELS + Extraction_batch + PCR_batch, data=data.frame(sample_data(pre_bdi_levels_rare_fil)))

allocation_all_fil<-subset_samples(depression2017_fil, Allocation_All %in% c("PreA", "PreB", "PostA", "PostB", "Con"))
allocation_all_ord<-ordinate(allocation_all_fil, method="PCoA", "wunifrac")
plot_ordination(allocation_all_fil, allocation_all_ord, color="Allocation_All")+stat_ellipse()

#beta-diversity pre and post graph 
allocation_groups<-subset_samples(depression2017_fil, Allocation_Groups %in% c("PreA", "PreB", "PostA", "PostB"))
allocation_groups_ord<-ordinate(allocation_groups, method="PCoA", "wunifrac")
plot_ordination(allocation_groups, allocation_groups_ord, color="Allocation_Groups")+stat_ellipse()+facet_wrap(~Allocation)+labs(title="Weighted UniFrac PCoA Allocation A and B; Pre and Post")

allocation_a<-subset_samples(depression2017_fil, Allocation_Groups %in% c("PreA","PostA"))
allocation_a_ord<-ordinate(allocation_a, method="PCoA", "wunifrac")
plot_ordination(allocation_a, allocation_a_ord, color="Allocation_Groups")+stat_ellipse()+labs(title="Weighted Unifrac PCoA Pre and Post Depressed Participant \nSamples Probiotic Treatment")

allocation_b<-subset_samples(depression2017_fil, Allocation_Groups %in% c("PreB","PostB"))
allocation_b_ord<-ordinate(allocation_b, method="PCoA", "wunifrac")

plot_ordination(allocation_b, allocation_b_ord, color="Allocation_Groups")+stat_ellipse()+labs(title="Weighted Unifrac PCoA Pre and Post Depressed Participant \nSamples Placebo Treatment")

```



```{r Plotting}

#Transform an OTU table from counts to proportions between 0 and 1.
depression2017_rel_1<-transform_sample_counts(depression2017_fil, function(x) x/sum(x)*100)

depression2017_rel_2<-transform_sample_counts(depression2017_rare, function(x) x/sum(x)*100)


depression2017_rel<-transform_sample_counts(depression2017_1, function(x) x/sum(x)*100)
                                            
#I use the psmelt function to get the data into a form that ggplot likes, dplyr to filter that data I want, and ggplot to do the plotting
#Melt data into a "longâ€ form. I used the OTU table rarefied to 28170 and converted to relative abundance.
depression_smelt<-psmelt(depression2017_rel_1)


depression_smelt_rare<-psmelt(depression2017_rel_2)

depression_smelt_1<-psmelt(depression2017_rel)
 ##Warning message:
##In psmelt(depression2017_rel_1) : The sample variables: 
##Sample
 ##have been renamed to: 
##sample_Sample
##to avoid conflicts with special phyloseq plot attribute names.

head(depression_smelt)


#Look for Bacillus OTUs in data since it was lost from mock community.
bacillus_otus<-depression_smelt %>% filter(Genus == "g__Bacillus") %>%
  ggplot(aes(x=Sample,y=Abundance))+geom_bar(stat="identity")+facet_grid(.~Participant, scales="free")
  
fp_otus<-depression_smelt %>% filter(Species=="s__prausnitzii")

ggplot(fp_otus, aes(x=Sample,y=Abundance))+geom_bar(stat="identity")+facet_grid(.~Participant, scales="free")

ggplot(fp_otus, aes(x=Participant,y=Abundance))+geom_boxplot(stat = "boxplot")

fp_otus %>% group_by(SampleID) %>% mutate(sum_abund=sum(Abundance)) %>%
  distinct(SampleID, .keep_all=TRUE) %>%
  ggplot(aes(x=Participant,y=Abundance))+geom_boxplot(stat = "boxplot")

f_otus<-depression_smelt %>% filter(Genus=="g__Faecalibacterium")

ggplot(f_otus, aes(x=Species, y=Abundance, fill=Species))+geom_bar(stat="identity")

#Let's plot the overall taxa by group (average relative abundance per group) and colour by phyla
#plot every sample by phyla, faceting into disease groups, 
phyla_plot<-depression_smelt %>% group_by(sample_Sample,Phylum) %>%
  mutate(abund=sum(Abundance)) %>% distinct(Phylum, .keep_all=TRUE) %>% select(sample_Sample, Phylum, Abundance, abund, Participant) %>% arrange(sample_Sample,Phylum)

phyla_plot_rare<-depression_smelt_rare %>% group_by(sample_Sample,Phylum) %>%
  mutate(abund=sum(Abundance)) %>% distinct(Phylum, .keep_all=TRUE) %>% select(sample_Sample, Phylum, Abundance, abund, Participant) %>% arrange(sample_Sample,Phylum)

ggplot(phyla_plot_rare, aes(x=sample_Sample, y=abund, fill=Phylum))+geom_bar(stat="identity") + facet_grid(PostA_B~Participant, scales="free")

probiotics<-depression_smelt_rare %>% filter(Genus %in% c("g__Bifidobacterium","g__Lactobacillus", "g__Lacotococcus")) %>% group_by(SampleID.1, Genus) %>% mutate(sum_abund= sum(Abundance)) %>% filter(Participant %in% c("Pre","Post")) %>% filter(SampleID.1!="P46.pre")%>% filter(Allocation_Groups %in% c("PostA", "PreA"))

probiotics$Participant<-factor(probiotics$Participant, levels=c("Pre","Post"))
probiotics$Allocation_Groups<-factor(probiotics$Allocation_Groups, levels=c("PreA","PostA"))

probiotics %>% filter(Pre_ProA%in% c("A","pre")) %>%
  ggplot(aes(x=sample_Sample,y=sum_abund, fill=Genus))+geom_bar(stat="identity")+facet_grid(.~Participant)+labs(title="Group A pre and post", y= "% relative abundance", x="Sample")+
  theme(axis.text.x = element_text(angle=90, size=5))

probiotics_lacto<-depression_smelt_rare %>% filter(Genus %in% c("g__Lacotococcus"))

ggplot(probiotics, aes(x=Participant_Group, y=sum_abund, fill=Allocation_Groups))+geom_bar(stat="identity", position="dodge")+labs(title="Pre and Post Probiotic Treatment Baseline\nDepressed Participants Probiotic Levels", y="% Relative Abundance", x="Sample",fill="Pre and Post")+theme(axis.text.x=element_text(angle=90, size=8))+facet_grid(.~Genus)+coord_fixed(ratio=155, expand = TRUE)+scale_fill_manual(values=c("deepskyblue3","red3"), labels=c("Pre Probiotic","Post Probiotic"))


probiotics_bif<- depression_smelt2 %>% filter(Genus %in% c("g__Bifidobacterium")) %>% group_by(Sample, Genus) %>% mutate(sum_abund= sum(Abundance)) %>% select(OTU,Sample,Abundance,Participant,Participant_Group,ProA_B,Pre_ProA ,Pre_PostB,Genus,sum_abund, Allocation_All, Allocation_Groups) %>% filter(Participant %in% c("Pre","Post")) %>% filter(Pre_ProA!="NA")%>% filter(Participant_Group!="P46")


ggplot(probiotics_bif, aes(x=Participant_Group, y=Abundance, fill=Allocation_Groups)) + geom_bar(stat="identity", position="dodge")+labs(title="Bifidobacterium Pre and Post Probiotic Treatment", y= "Abundance", x="Sample")+theme(axis.text.x = element_text(angle=90, size=10))

probiotics_bif$Allocation_Groups<-factor(probiotics_bif$Allocation_Groups, levels = c("PreA", "PostA"))

ggplot(probiotics_bif, aes(x=Participant_Group, y=Abundance, fill=factor(Allocation_Groups))) + geom_bar(stat="identity", position="dodge")+labs(title="Bifidobacterium Pre and Post Probiotic Treatment", y= "Abundance", x="Sample", fill="Pre vs Post")+theme(axis.text.x = element_text(angle=90, size=10))

probiotics$Participant<-factor(probiotics$Participant, levels=c("Pre","Post"))


probiotics %>% filter(Pre_ProA%in% c("A","pre")) %>% filter(SampleID.1!=("P46.post"))%>% ggplot(aes(x=Participant_Group,y=sum_abund, fill=Participant_g))+geom_bar(stat="identity", position=position_dodge())+labs(title="Pre and Post Probiotic Treatment Depressed Samples", y= "% Relative Abundance", x="Sample")+ theme(axis.text.x = element_text(angle=90, size=5))


+facet_grid(.~Participant)

probiotics %>% filter(Pre_PostB%in% c("B","pre")) %>% filter(SampleID)
  ggplot(aes(x=sample_Sample,y=sum_abund, fill=Genus))+geom_bar(stat="identity")+facet_grid(.~Participant)+labs(title="Group B pre and post", y= "% relative abundance", x="Sample")+
  theme(axis.text.x = element_text(angle=90, size=5))
 
proA<-probiotics %>% filter(Pre_ProA %in% c("A","pre"))

kruskal.test(Abundance ~ Pre_ProA, proA)

proB<-probiotics %>% filter(Pre_PostB %in% c("B","pre"))
kruskal.test(Abundance ~ Pre_PostB, proB)

ggplot(probiotics, aes(x=sample_Sample, y=Abundance, fill=Genus))+geom_bar(stat="identity") + facet_grid(.~ProA_B, scales="free")

ggplot(probiotics, aes(x=ProA_B, y=sum_abund, fill=Genus))+geom_boxplot() 
                                              
roseburia<-depression_smelt_rare %>% filter(OTU=="518438") %>% arrange(desc(Abundance))

depression_smelt_rare_counts<-psmelt(depression2017_rare)
depression_smelt_fil_counts<-psmelt(depression2017_fil)

roseburia_counts<-depression_smelt_rare_counts %>% filter(OTU=="518438") %>% arrange(desc(Abundance))

head(roseburia)
ggplot(roseburia_counts, aes(x=Abundance, y=CogReact.Total.Score))+geom_point()

lacno_counts<-depression_smelt_rare_counts %>% filter(OTU=="205148") %>% arrange(desc(Abundance)) %>% filter(Participant!="Con")

ggplot(lacno_counts, aes(x=Abundance, y=CogReact.Total.Score))+geom_point()+geom_smooth(method=lm)


roseburia_counts_DepDass<-depression_smelt_rare_counts %>% filter(OTU=="518438") %>% arrange(desc(Abundance)) %>% filter(Participant!="Post")

roseburia_DepDass<-depression_smelt_rare %>% filter(OTU=="518438") %>% arrange(desc(Abundance)) %>% filter(Participant!="Post")

head(roseburia_counts_DepDass)
tail(roseburia_counts_DepDass)

ggplot(roseburia_counts_DepDass, aes(x=Abundance, y=DEP_DASS))+geom_point(aes(colour=Participant)) + labs(title="DEP_Dass Correlation to Rosburia 518438 in Baseline \nDepressed and Control Samples")+geom_smooth(method=lm)+scale_colour_manual(values=c("green","red", labels=c("Control", "Baseline Depressed")))


ruminococcus_counts_DepDass<-depression_smelt_rare_counts %>% filter(OTU=="4341886") %>% arrange(desc(Abundance)) %>% filter(Participant!="Post")
ggplot(ruminococcus_counts_DepDass, aes(x=Abundance, y=DEP_DASS))+geom_point() + labs(title="Dep_Dass Pre vs. Con Ruminococcus 4341886")+geom_smooth(method=lm)


clostridiales_counts_cogn<-depression_smelt_rare_counts %>% filter(OTU=="358594") %>% arrange(desc(Abundance)) %>% filter(Participant!="Post")
ggplot(clostridiales_counts_cogn, aes(x=Abundance, y=CogReact.Total.Score))+geom_point() + labs(title="CogReact Pre vs. Con Clostridiales 358594")+geom_smooth(method=lm)

Ruminococcaceae_counts_cogn<-depression_smelt_rare_counts %>% filter(OTU=="295748")+filter(PRE_BDI_LEVELS%in% c("mild", "severe"))

ggplot(Ruminococcaceae_counts_cogn, aes (x=PRE_BDI_LEVELS, y=Abundance))+geom_boxplot()+geom_point()

Top10OTUs<-names(sort(taxa_sums(depression2017_rare),TRUE))[1:10]

top_10<-depression_smelt_rare %>% filter(OTU %in% Top10OTUs)
ggplot(top_10, aes(x=OTU, y=Abundance, fill=Participant))+geom_bar(stat="identity", position="dodge")

Top100OTUs<-names(sort(taxa_sums(depression2017_rare),TRUE))[1:100]
top_100<-depression_smelt_rare_counts %>% filter(OTU %in% Top100OTUs)

ggplot(top_10, aes(x=OTU, y=Abundance, fill=Participant))+geom_boxplot()

```


```{r Writing to a textfile}
write.table((depression_smelt_1 %>% distinct(OTU)), file="all_otus1.txt", sep="\t")
```

```{r}

Top100OTUs<-names(sort(taxa_sums(depression2017_rare),TRUE))[1:100] 

top_100<-depression_smelt_rare %>% filter(OTU %in% Top100OTUs)

top_100_PreCon<-top_100 %>% filter(Participant!="Post")

top_100_gnavus<-top_100_PreCon %>% filter(OTU=="360015") %>% filter(PRE_BDI_LEVELS!="NA")

ggplot(top_100_gnavus, aes(x=OTU, y=Abundance, fill=PRE_BDI_LEVELS)) %>% +geom_bar(stat="identity", position="dodge")

ggplot(top_100_gnavus, aes(x=OTU, y=Abundance, fill=PRE_BDI_LEVELS)) %>% +geom_boxplot()

ggplot(top_100, aes(x=Phylum, y=Abundance, fill=Participant))+geom_bar(stat="identity")

top_100_gnavus$PRE_BDI_LEVELS<-factor(top_100_gnavus$PRE_BDI_LEVELS, levels=c("mild", "moderate", "severe", "Con"))

ggplot(top_100_gnavus, aes(x=Abundance, y=DEP_DASS))+geom_point(aes(colour=PRE_BDI_LEVELS)) + labs(title="DEP_DASS correlation with R. gnavus 360015", colour="BDI Severity Levels")+geom_smooth(method=lm)+scale_colour_manual(values=c("darkgray","blue","yellow","red"), labels=c("Mild", "Moderate", "Severe", "Control"))

ggplot(top_100_gnavus, aes(x=Abundance, y=DEP_DASS))+geom_point()

depression_smelt_rare_counts %>% mutate(meanAbund= mean(Abundance)) %>% filter(Family=="f__Ruminococcaceae") %>% filter(PRE_BDI_LEVELS!=("NA")) %>% filter(PRE_BDI_LEVELS!=("moderate")) %>%ggplot(aes(x=PRE_BDI_LEVELS, y=meanAbund, fill=PRE_BDI_LEVELS))+ geom_boxplot()+labs(title="Mean Abundance of f_Ruminococcaceae Severe & Mild BDI")

depression_smelt_rare_counts %>% mutate(meanAbund= mean(Abundance)) %>% filter(OTU=="295748") %>% filter(PRE_BDI_LEVELS!=("NA")) %>% filter(PRE_BDI_LEVELS!=("moderate")) %>%ggplot(aes(x=PRE_BDI_LEVELS, y=Abundance))+ geom_point()+labs(title="Abundance of OTU 295748 f_Ruminococcaceae in Severe & Mild \nBDI Baseline Depressed Samples", x="BDI Severity Level", fill="BDI Severity Level")


```

```{r Results Figures}

##Mock Community

##Rarefaction Curve
t_otu_depression2017<-as.matrix(t(otu_table(depression2017_1)))
curve<-rarecurve(t_otu_depression2017, step=500, label=FALSE, xlim=c(0,50000), ylim=c(0,2000), ylab = "OTU Count", xaxt="n")

axis(1)
abline(v=28170, col="red")


##Phyla Present PreCon (After rare filtering)

phyla_plot<-depression_smelt %>% filter(Participant %in% c("Pre", "Con"))

phyla_plot_av<-depression_smelt %>% 
  filter(Participant %in% c("Pre", "Con")) %>% select(Participant, SampleID.1, Phylum, Abundance) %>% group_by(Participant, Phylum, SampleID.1) %>% mutate(Abund = Abundance) %>% summarise(sum_p=sum(Abundance)) %>%
  summarise(av_abund=mean(sum_p), std=sd(sum_p))

phyla_bacte<-depression_smelt_rare %>% filter(Participant %in% c("Pre", "Con")) %>% filter(Phylum=="p__Bacteroidetes") %>% select(Phylum, Abundance)

phyla_bacte %>% group_by(Phylum, Abundance) %>% summarise(meanAbund=mean(Abundance), sd=sd(Abundance))

phyla_firm<-depression_smelt_rare %>% filter(Participant %in% c("Pre", "Con")) %>% filter(Phylum=="p__Firmicutes") %>% select(Participant, Phylum, Abundance, SampleID.1)


ggplot(phyla_plot_av, aes(x=Participant, y=av_abund, fill=Phylum))+geom_bar(stat="identity")+scale_x_discrete(labels=c("Controls", "Baseline Depressed"))+scale_fill_manual(values=c("red2", "orange1", "yellow1", "limegreen", "deepskyblue2","purple1", "hotpink"), labels=c("Actinobacteria", "Bacteroidetes", "Firmicutes", "Proteobacteria", "Synergistetes", "TM7", "Verrucomicrobia"))

pre_con_labels<-c('Con'="Controls", 'Pre'="Depressed")

ggplot(phyla_plot, aes(x=Sample, y=Abundance, fill=Phylum))+geom_bar(stat="identity")+scale_fill_manual(values=c("red2", "orange1", "yellow1", "limegreen", "deepskyblue2","purple1", "hotpink"), labels=c("Actinobacteria", "Bacteroidetes", "Firmicutes", "Proteobacteria", "Synergistetes", "TM7", "Verrucomicrobia"))+facet_grid(.~Participant, labeller=as_labeller(pre_con_labels), scales="free")+theme(legend.key.size=unit(1, "line"), axis.text.x=element_text(angle=90, hjust=1, size=5))+labs(y="% Relative Abundance")


##PreCon AlphaDiv
plot_richness(dep_con_richness, x="Dep_Con", measures=c("Observed"))+ geom_boxplot()+xlab("Group")+ylab("Diversity")+scale_x_discrete(limits=c("Dep", "Con"), labels=c("Depressed", "Controls"))+theme(axis.text.x=element_text(angle=360))+labs(y="Richness")


observed_rich<-c('Observed'="Observed Richness")

##PreCon BetaDiv
plot_ordination(pre_con_rare_fil,pre_con_ord, color="Participant") + stat_ellipse()+scale_color_manual(values=c("limegreen", "red2"), labels=c("Controls", "Depressed"))+theme(legend.key.size=unit(1, "line"))+labs(color="Group")

##PreCon DepDass Roseburia 
depdass_roseburia<-ggplot(roseburia_DepDass, aes(x=Abundance, y=DEP_DASS))+geom_point(aes(color=Participant))+geom_smooth(method=lm)

depdass_roseburia + scale_colour_manual(values=c("green","red"), labels=c("Controls", "Depressed"))+labs(color="Group", y="DASS Depression Score", x="% Relative Abundance")+theme(legend.key.size=unit(1,"line"))

roseburia_dep<- roseburia_DepDass %>% select(Abundance, DEP_DASS, Participant, OTU)

roseburia_dep%>% summarise(sum_Abund=sum(Abundance), std=sd(Abundance))

##SevereMild Ruminoc
severemild_rumino<-depression_smelt_rare %>% mutate(meanAbund= mean(Abundance)) %>% filter(OTU=="295748") %>% mutate(stddev= sd(Abundance)) %>%filter(PRE_BDI_LEVELS %in% c("severe","mild")) %>% select(Abundance, OTU, meanAbund, stddev, PRE_BDI_LEVELS)

ggplot(severemild_rumino, aes(x=PRE_BDI_LEVELS, y=Abundance))+ geom_point()+labs(x="BDI Severity Level", fill="BDI Severity Level", y="% Relative Abundance")+geom_boxplot()+scale_x_discrete(labels=c("Mild","Severe")) 

severemild_rumino
severemild_rumino %in% count(Abundance, 0)
  
##PrePost AlphaDiv
allocation_a_richness<-subset_samples(depression2017_fil, Allocation_Groups %in% c("PreA", "PostA"))

obs_a_plot<-plot_richness(allocation_a_richness, x="Allocation_Groups", measures=c("Observed"))+ geom_boxplot()+xlab("Group")+ylab("Diversity")+scale_x_discrete(limits=c("PreA", "PostA"), labels=c("Pre Probiotic", "Post Probiotic"))+theme(axis.text.x=element_text(angle=360))+labs(y="Richness")

allocation_b_richness<-subset_samples(depression2017_fil, Allocation_Groups %in% c("PreB", "PostB"))
obs_b_plot<-plot_richness(allocation_b_richness, x="Allocation_Groups", measures=c("Observed"))+ geom_boxplot()+xlab("Group")+ylab("Diversity")+scale_x_discrete(limits=c("PreB", "PostB"), labels=c("Pre Placebo", "Post Placebo"))+theme(axis.text.x=element_text(angle=360))+labs(y="Richness")

grid.arrange(obs_a_plot, obs_b_plot, ncol=2)
g <- arrangeGrob(obs_a_plot, obs_b_plot, ncol=2)
 ggsave(file="whatever.pdf", g)



##PrePost BetaDiv (line 287)
plot_ordination(allocation_groups, allocation_groups_ord, color="Allocation_Groups")+stat_ellipse()+facet_wrap(~Allocation)+labs(title="Weighted UniFrac PCoA Allocation A and B; Pre and Post")

allocation_a<-subset_samples(depression2017_fil, Allocation_Groups %in% c("PreA","PostA"))
allocation_a_ord<-ordinate(allocation_a, method="PCoA", "wunifrac")

a_beta<-plot_ordination(allocation_a, allocation_a_ord, color="Allocation_Groups")+stat_ellipse()+labs(color="Probiotics")+scale_color_manual(values=c("deepskyblue2","orange"), limits=c("PreA", "PostA"), labels=c("Pre Probiotic","Post Probiotic"))

allocation_b<-subset_samples(depression2017_fil, Allocation_Groups %in% c("PreB","PostB"))
allocation_b_ord<-ordinate(allocation_b, method="PCoA", "wunifrac")

b_beta<-plot_ordination(allocation_b, allocation_b_ord, color="Allocation_Groups")+stat_ellipse()+labs(color="Placebo")+scale_color_manual(values=c("deepskyblue2","orange"), limits=c("PreB", "PostB"), labels=c("Pre Placebo","Post Placebo"))

grid.arrange(a_beta, b_beta, nrow=2)
g <- arrangeGrob(a_beta, b_beta, nrow=2)
 ggsave(file="Beta (PrePost).pdf", g)

##Top100 Dep_Dass Severity Correlation
top_100_gnavus$PRE_BDI_LEVELS<-factor(top_100_gnavus$PRE_BDI_LEVELS, levels=c("mild", "moderate", "severe", "Con"))

ggplot(top_100_gnavus, aes(x=Abundance, y=DEP_DASS))+geom_point(aes(colour=PRE_BDI_LEVELS)) + labs(colour="BDI Severity\nLevels", y="DASS Depression Score", x="% Relative Abundance")+geom_smooth(method=lm)+scale_colour_manual(values=c("dodgerblue1","orange1","red2","limegreen"), labels=c("Mild", "Moderate", "Severe", "Control"))+theme(legend.key.size=unit(1,"line"))


##Probiotics 
ggplot(probiotics, aes(x=Participant_Group, y=sum_abund, fill=Allocation_Groups))+geom_bar(stat="identity", position="dodge")+labs(x="Sample",fill="Probiotics",y="% Relative Abundance")+theme(axis.text.x=element_text(angle=90, size=7), legend.key.size=unit(1,"line"))+facet_grid(.~Genus)+scale_fill_manual(values=c("deepskyblue3","orange"), labels=c("Pre Probiotic","Post Probiotic"))

probiotics$Allocation_Groups<-factor(probiotics$Allocation_Groups, levels=c("PreA", "PostA"))




```

```{r, ANOVA pre scores}
pre_data<-data.frame(sample_data(allocation_groups)[sample_data(allocation_groups)$Allocation_Groups_All %in% c("PreA","PreB")])
anova(lm(BDI~Allocation_Groups_All, data=pre_data))
anova(lm(DEP_DASS~Allocation_Groups_All, data=pre_data))
anova(lm(ANX_DASS~Allocation_Groups_All, data=pre_data))
anova(lm(STR_DASS~Allocation_Groups_All, data=pre_data))
anova(lm(CogReact.Total.Score~Allocation_Groups_All, data=pre_data))
anova(lm(Age~Allocation_Groups_All, data=pre_data))


precon<-subset_samples(depression2017_fil, Dep_Con %in% c("Dep","Con"))
precon_data<-data.frame(sample_data(precon)[sample_data(precon)$Dep_Con %in% c("Dep","Con")])

anova(lm(Age~Dep_Con, data=precon_data))
anova(lm(BDI~Dep_Con, data=precon_data))
anova(lm(DEP_DASS~Dep_Con, data=precon_data))
anova(lm(ANX_DASS~Dep_Con, data=precon_data))
anova(lm(STR_DASS~Dep_Con, data=precon_data))
anova(lm(CogReact_Score~Dep_Con, data=precon_data))
```

```{r Significant Test Probiotics}

probiotics<-depression_smelt_rare %>% filter(Genus %in% c("g__Bifidobacterium","g__Lactobacillus", "g__Lacotococcus")) %>% group_by(SampleID.1, Genus) %>% mutate(sum_abund= sum(Abundance)) %>% filter(Participant %in% c("Pre","Post")) %>% filter(SampleID.1!="P46.pre")%>% filter(Allocation_Groups %in% c("PostA", "PreA"))

probiotics_bifido<-probiotics %>% filter(Genus %in% c("g__Bifidobacterium")) %>% filter(Allocation_Groups %in% c("PreA", "PostA"))

probiotics_lactobac<-probiotics %>% filter(Genus %in% c("g__Lactobacillus")) %>% filter(Allocation_Groups %in% c("PreA", "PostA"))

probiotics_lactococ<-probiotics %>% filter(Genus %in% c("g__Lacotococcus")) %>% filter(Allocation_Groups %in% c("PreA", "PostA"))

kruskal.test(Abundance ~ Pre_ProA, probiotics_bifido)
kruskal.test(Abundance ~ Pre_ProA, probiotics_lactobac)
kruskal.test(Abundance ~ Pre_ProA, probiotics_lactococ)

probiotics_lacto_pre<-depression_smelt_rare %>% filter(Genus %in% c("g__Lactobacillus")) %>% filter(SampleID.1!="P46.pre")%>% filter(Allocation_Groups %in% c("PreA")) %>% select(Abundance, Allocation_Groups, Participant, Genus, OTU)

probiotics_lacto_post<-depression_smelt_rare %>% filter(Genus %in% c("g__Lactobacillus")) %>% filter(SampleID.1!="P46.pre")%>% filter(Allocation_Groups %in% c("PostA")) %>% select(Abundance, Allocation_Groups, Participant, Genus, OTU)

probiotics_lacto_pre %>% summarise(meanAbund=mean(Abundance), sd=sd(Abundance))
probiotics_lacto_post %>% summarise(meanAbund=mean(Abundance), sd=sd(Abundance))

ggplot(probiotics_lacto, aes(x=Abundance, y=Allocation_Groups))+geom_boxplot()

#placebo 
placebo<-depression_smelt_rare %>% filter(Genus %in% c("g__Bifidobacterium","g__Lactobacillus", "g__Lacotococcus")) %>% group_by(SampleID.1, Genus) %>% mutate(sum_abund= sum(Abundance)) %>% filter(Participant %in% c("Pre","Post")) %>% filter(SampleID.1!="P46.pre")%>% filter(Allocation_Groups %in% c("PostA", "PreA"))

```

```{r StdDev for Mild/Severe}

severemild<-depression_smelt_rare %>% filter(Participant=="Pre") %>% filter(Severe_Mild %in% c("severe", "mild")) %>% filter(OTU=="295748") %>% group_by(Severe_Mild, OTU) %>% summarise(meanAbund=mean(Abundance), sd=sd(Abundance))


severemild_counts<-depression_smelt_fil_counts %>% filter(Participant=="Pre") %>% filter(Severe_Mild %in% c("severe", "mild")) %>% filter(OTU=="295748") %>% group_by(Severe_Mild, OTU) %>% summarise(meanAbund=mean(Abundance), sd=sd(Abundance))
  

kruskal.test(Severe_Mild ~ Abundance, data=severemild)
```

```{r Mock Community}

#Check sequencing controls - mock communtiy
#Look at blast results of mock seqs against reference seqs.  
mock_blast<-read.delim(file="data/mock_seqs_full_blast.txt", header=TRUE, sep="\t")

mock_blast %>% count(sseqid, pident) %>% 
  arrange(desc(pident))%>% arrange(sseqid) %>% group_by(sseqid) %>%
  dplyr::summarise(n_distinct(pident))
#Summarise the number of hits with a distinct percent identity for each taxa in the mock community
mock_summary<-mock_blast %>% group_by(sseqid, pident) %>%
  arrange(sseqid, desc(pident)) %>%
  summarise(n=n()) %>%
  arrange(sseqid, desc(pident)) %>% group_by(sseqid) %>%
  summarise(counts=sum(n))

mock_98<-mock_blast %>% 
  filter(pident>=98.000) %>%
  arrange(sseqid,desc(pident)) %>% group_by(sseqid, pident) %>%
  summarise(n=n()) %>%
  group_by(sseqid) %>%
  summarise(counts=sum(n))
  
mock_97<-mock_blast %>% 
  filter(pident>=97.000) %>%
  arrange(sseqid,desc(pident)) %>% group_by(sseqid, pident) %>%
  summarise(n=n()) %>%
  group_by(sseqid) %>%
  summarise(counts=sum(n))

#Calculate proportion of seqs with < 97% identity to reference
less_97<-mock_blast %>%
  filter(pident<97) %>%
  nrow()/nrow(mock_blast)
less_99<- mock_blast %>%
  filter(pident<99) %>%
  nrow()/nrow(mock_blast)  

#Calculate observed vs expected frequency of taxa for hits > 97% identity to a reference.

#observed<- mock_97 %>% group_by(queryseqid,sseqid) %>% mutate(count=1)%>% group_by(sseqid) %>% summarise(counts=sum(count)) %>% mutate(observed=counts/sum(counts)*100)

##wasn't working so did the code below:
observed<- mock_blast %>%filter(pident<97) %>% group_by(queryseqid,sseqid) %>% mutate(count=1)%>% group_by(sseqid) %>% summarise(counts=sum(count)) %>% mutate(observed=counts/sum(counts)*100)

expected<-c(15.7,10.4,10.0,18.8,15.9,4.6,11.3,13.3)

observed_expected<-cbind(observed,expected) %>% select(sseqid, observed, expected)
observed_expected$sseqid<-gsub("_16S[_2]*","", observed_expected$sseqid)

melt(observed_expected) %>% 
  ggplot(aes(x=sseqid,y=value, fill=variable))+geom_bar(stat="identity", position="dodge") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+
  labs(x="Mock Community Species", y="% Relative Abundance")+
  theme(legend.title=element_blank())

observed_expected<-observed_expected %>% mutate(fc=observed/expected) %>% 
  mutate(log2fc=log2(fc))

observed_expected %>% select(sseqid, observed, expected) %>% melt() %>% 
  ggplot(aes(x=sseqid,y=value, fill=variable))+geom_bar(stat="identity", position="dodge") + theme(axis.text.x = element_text(angle = 45, hjust = 1))+labs(x="Mock Community Species", y="% Relative Abundance", fill="")+scale_fill_manual(values=c("deepskyblue","red3"), labels=c("Observed", "Expected"))

```

```{r ANCOVA}


#BDI
ancova_data<- data.frame(sample_data(depression2017)) %>% 
  filter(Depression=="Y") %>%
  mutate(patient=gsub("([P0-9]*).[postre]*","\\1",SampleID.1)) %>%
  arrange(SampleID.1) %>% 
  select (patient, SampleID.1, BDI,Probiotics,Participant,ProA_B) %>%
  filter(Probiotics %in% c("B","A","pre")) %>%
  do(na.locf(.)) %>% select(patient, BDI,Participant,ProA_B) %>%
  dcast(patient + ProA_B ~ Participant, value.var="BDI") %>%
  filter(patient!="P62")
ancova_data$Post<-as.numeric(ancova_data$Post)
ancova_data$Pre<-as.numeric(ancova_data$Pre)
ggplot(ancova_data, aes(x=Post, y=Pre, colour=ProA_B, shape=ProA_B))+
  geom_point() +
  geom_smooth(method=lm, se=FALSE, fullrange=TRUE)

anova(lm(Post~Pre*ProA_B, data=ancova_data))

anova(lm(Post~Pre + ProA_B, data=ancova_data))

#Depression DASS
ancova_data_dep<- data.frame(sample_data(depression2017)) %>% 
  filter(Depression=="Y") %>%
  mutate(patient=gsub("([P0-9]*).[postre]*","\\1",SampleID.1)) %>%
  arrange(SampleID.1) %>% 
  select (patient, SampleID.1, DEP_DASS,Probiotics,Participant,ProA_B) %>%
  filter(Probiotics %in% c("B","A","pre")) %>%
  do(na.locf(.)) %>% select(patient, DEP_DASS,Participant,ProA_B) %>%
  dcast(patient + ProA_B ~ Participant, value.var="DEP_DASS") %>%
  filter(patient!="P62")
ancova_data_dep$Post<-as.numeric(ancova_data_dep$Post)
ancova_data_dep$Pre<-as.numeric(ancova_data_dep$Pre)

anova(lm(Post~Pre*ProA_B, data=ancova_data_dep))

anova(lm(Post~Pre + ProA_B, data=ancova_data_dep))

#Anxiety DASS 
ancova_data_anx<- data.frame(sample_data(depression2017)) %>% 
  filter(Depression=="Y") %>%
  mutate(patient=gsub("([P0-9]*).[postre]*","\\1",SampleID.1)) %>%
  arrange(SampleID.1) %>% 
  select (patient, SampleID.1, ANX_DASS,Probiotics,Participant,ProA_B) %>%
  filter(Probiotics %in% c("B","A","pre")) %>%
  do(na.locf(.)) %>% select(patient, ANX_DASS,Participant,ProA_B) %>%
  dcast(patient + ProA_B ~ Participant, value.var="ANX_DASS") %>%
  filter(patient!="P62")
ancova_data_anx$Post<-as.numeric(ancova_data_anx$Post)
ancova_data_anx$Pre<-as.numeric(ancova_data_anx$Pre)

anova(lm(Post~Pre*ProA_B, data=ancova_data_anx))

anova(lm(Post~Pre + ProA_B, data=ancova_data_anx))

#Stress DASS 
ancova_data_str<- data.frame(sample_data(depression2017)) %>% 
  filter(Depression=="Y") %>%
  mutate(patient=gsub("([P0-9]*).[postre]*","\\1",SampleID.1)) %>%
  arrange(SampleID.1) %>% 
  select (patient, SampleID.1, STR_DASS,Probiotics,Participant,ProA_B) %>%
  filter(Probiotics %in% c("B","A","pre")) %>%
  do(na.locf(.)) %>% select(patient, STR_DASS,Participant,ProA_B) %>%
  dcast(patient + ProA_B ~ Participant, value.var="STR_DASS") %>%
  filter(patient!="P62")
ancova_data_str$Post<-as.numeric(ancova_data_str$Post)
ancova_data_str$Pre<-as.numeric(ancova_data_str$Pre)

anova(lm(Post~Pre*ProA_B, data=ancova_data_str))

anova(lm(Post~Pre + ProA_B, data=ancova_data_str))

#Cognitive Reactivity 
ancova_data_cog<- data.frame(sample_data(depression2017)) %>% 
  filter(Depression=="Y") %>%
  mutate(patient=gsub("([P0-9]*).[postre]*","\\1",SampleID.1)) %>%
  arrange(SampleID.1) %>% 
  select (patient, SampleID.1, CogReact_Score,Probiotics,Participant,ProA_B) %>%
  filter(Probiotics %in% c("B","A","pre")) %>%
  do(na.locf(.)) %>% select(patient, CogReact_Score,Participant,ProA_B) %>%
  dcast(patient + ProA_B ~ Participant, value.var="CogReact_Score") %>%
  filter(patient!="P62")
ancova_data_cog$Post<-as.numeric(ancova_data_cog$Post)
ancova_data_cog$Pre<-as.numeric(ancova_data_cog$Pre)

anova(lm(Post~Pre*ProA_B, data=ancova_data_cog))

anova(lm(Post~Pre + ProA_B, data=ancova_data_cog))
```

