---
title: "Depression_Microbiome_Data"
author: "Bahia Chahwan"
date: "22/08/2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r load packages}
library("phyloseq")
library("ggplot2")
library("vegan")
library("reshape2")
library("tidyverse")
```

#Setting up in R
```{r Setting up in R}

#otu clustered and chimeras removed
biom<-"data/otu_table_final1_json.biom"

#didn't run this as it wouldnt work
map<-read.table("data/mapping_depression_newR.txt", header=TRUE, row.names=1)

#had to run this comand as the other one (^) wouldnt work for some reason 
map<-read.table(file.choose("data/mapping_depression_newR.txt"), sep="\t", header=TRUE, row.names=1)

#Normally would just use the read_tree command, but this particular tree is a bit different to ususal.
tree<-read_tree("data/rep_set.tre")

## Import Data to PhyloSeq
depression2017<-import_biom(biom, treefilename=tree)

depression_map<-sample_data(map)

depression2017<-merge_phyloseq(depression2017,depression_map)

#18696 taxa and 142 samples
depression2017
head(depression2017)
colnames(tax_table(depression2017)) = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species")

```


#Cleaning up data
```{r Cleaning up data}

#Removed participant incorrectly enrolled and mock 
depression2017_1<-subset_samples(depression2017, !(SampleID.1 %in% c("P9.pre","P7.pre","P7.post", "mock")))

#18695 taxa and 138 samples
depression2017_1

#Rarefy to 28170 - lose p4.pre
depression2017_rare<-rarefy_even_depth(depression2017_1, sample.size=28170, rngseed=711, replace=FALSE, trimOTUs=TRUE)
##7 samples removed because they contained fewer reads than `sample.size`
##2364OTUs were removed because they are no longer present in any sample after random subsampling

#16331 taxa and 131 samples 
depression2017_rare


#filter out rare OTUs
depression2017_fil<- filter_taxa(depression2017_rare, function(x) sum(x)>=10 & sum(x>0)>0.25*length(x), TRUE)

#938 taxa and 131 samples
depression2017_fil
```


#Rarefaction Curve
```{r Rarefaction Curve}

#Create a rarefaction curve to find the appropraite place for rarefying your data. Uses vegan for calculating the curve, and base R for plotting. 
#Plot rarecurve
t_otu_depression2017<-as.matrix(t(otu_table(depression2017_1)))
curve<-rarecurve(t_otu_depression2017, step=500, label=FALSE, xlim= c(0,50000), ylim=c(0,2500), ylab = "OTU Count", xaxt="n")
axis(1, xaxp=c(0, 100000, 50))
curve
#place a line over the curve where you proppose to rarefy
abline(v=247, col="red")

```


#subsetting phyloseq objects
##havn't done this yet
```{r, subsetting phyloseq objects}

#You many want to remove certain samples or OTUs from the analysis.  There are several ways to do this in Phyloseq, all require creating a new Phyloseq object.
#First example, subset to samples that match a string in a column of the sample data (i.e. the mapping file)
practice2017_rare_controls<-subset_samples(practice2017_rare, Disease_status=="control")
sample_data(practice2017_rare_controls)
#Second example, subset to samples that match everything except the string in a column of the sample data
practice2017_rare_disease<-subset_samples(practice2017_rare, Disease_status!="control")
#Now check the sample data for your new object to make sure it did what you think it did.  In this example I'm only returning the column that lists Disease_status
sample_data(practice2017_rare_disease)$Disease_status
#Third example, subset to samples that match several strings in a column of the sample data
practice2017_rare_disease2<-subset_samples(practice2017_rare, Disease_status%in%c("mdd", "recovering_mdd"))
#And check the sample data as before.
sample_data(practice2017_rare_disease2)$Disease_status
#Forth example, subset to samples that match everything except several strings in a column of the sample data
practice2017_rare_controls2<-subset_samples(practice2017_rare, !(Disease_status%in%c("mdd", "recovering_mdd")))
#And check the sample data as before.
sample_data(practice2017_rare_controls2)$Disease_status
```

#Alpha diversity analysis
```{r, alpha diversity}
#participant (pre, post, con)

#Once you have rarefied your data, explore the level of alpha diversity.  Uses phyloseq plot function which is built on ggplot2.  The x value will determine how samples are grouped along the x axis.
plot_richness(depression2017_fil, x="Participant", measures=c("Observed", "Chao1","Shannon"))+
  geom_boxplot()+xlab("Group")+ylab("Diversity")

#Convert alpha diversity estimates to a data form we can use for statistical testing
depression_rare_richness<-as.matrix(estimate_richness(depression2017_fil, measures=c("Observed","Chao1","Shannon")))

##don't need this - row names are correct
#row.names(practice2017_rare_richness)<-row.names(sample_data(practice2017_rare))

depression_rare_richness<-cbind(depression_rare_richness,sample_data(depression2017_fil))

depression_rare_richness

#For comaparisons of alpha diversity of more than two groups, use the Kruskal-Wallis test 
#Test if alpha diversity (observed) is significantly different between Disease_status groups
kruskal.test(Observed ~ Participant, data=depression_rare_richness)
##Kruskal-wallis rank sum test, pvalue 0.75498, thus no significant difference between populations, null hypothesis assumes samples are from identical populations

#Now test if diversity is significantly different between controls and patients who were ever depressed (and recovering mdd groups (Original_status)
#only two groups here so use Man-Whitney U test
#null asserts that medians of two samples are indentical 
wilcox.test(Observed ~ Dep_Con, data=depression_rare_richness)
##data:  Observed by Dep_Con W = 769.5, p-value = 0.3755
##alternative hypothesis: true location shift is not equal to 0

wilcox.test(Observed ~ Con_A, data=depression_rare_richness)
##data:  Observed by Con_A W = 180, p-value = 0.4416


##Warning message:
##In wilcox.test.default(x = c(412, 401, 409, 445, 381, 434, 511,  :
  ##cannot compute exact p-value with ties

wilcox.test(Observed ~ Con_B, data=depression_rare_richness)
##data:  Observed by Con_B W = 216, p-value = 0.9297

wilcox.test(Observed ~ Pre_ProA, data=depression_rare_richness)
##data:  Observed by Pre_ProA W = 239, p-value = 0.654

wilcox.test(Observed ~ Pre_PostB, data=depression_rare_richness)
##data:  Observed by Pre_PostB W = 265.5, p-value = 0.5892

wilcox.test(Observed ~ ProA_B, data=depression_rare_richness)
##data:  Observed by ProA_B W = 203, p-value = 0.504

##PRE_BDI_LEVELS - severe, mild moderate
kruskal.test(Observed ~ PRE_BDI_LEVELS, data=depression_rare_richness)
##Kruskal-Wallis chi-squared = 3.1845, df = 2, p-value = 0.2035

```


#Beta Diversity
```{r Beta Diversity}

depression2017_wuf<-distance(depression2017_fil, "wunifrac", type="samples")
##Warning message:
##In UniFrac(physeq, weighted = TRUE, ...) :
  ##Randomly assigning root as -- 531539 -- in the phylogenetic tree in the data you provided.

depression2017_wuf_ord<-ordinate(depression2017_fil, method="PCoA", distance=depression2017_wuf)

plot_ordination(depression2017_fil, depression2017_wuf_ord, type="samples",color="Participant")

pre_con_rare_fil<-subset_samples(depression2017_fil, Participant %in% c("Pre","Con"))
pre_con_ord<-ordinate(pre_con_rare_fil, method="PCoA", "wunifrac")

plot_ordination(pre_con_rare_fil,pre_con_ord, color="Participant")

con_A_rare_fil<-subset_samples(depression2017_fil, Con_A %in% c("Con","A"))
con_A_ord<-ordinate(con_A_rare_fil, method="PCoA", "wunifrac")

plot_ordination(con_A_rare_fil,con_A_ord, color="Con_A")

con_B_rare_fil<-subset_samples(depression2017_fil, Con_B %in% c("Con","B"))
con_B_ord<-ordinate(con_B_rare_fil, method="PCoA", "wunifrac")

plot_ordination(con_B_rare_fil,con_B_ord, color="Con_B")

pre_proA_rare_fil<-subset_samples(depression2017_fil, Pre_ProA %in% c("pre","A"))
pre_proA_ord<-ordinate(pre_proA_rare_fil, method="PCoA", "wunifrac")

plot_ordination(pre_proA_rare_fil, pre_proA_ord, color="Pre_ProA")

pre_proB_rare_fil<-subset_samples(depression2017_fil, Pre_PostB %in% c("pre","B"))
pre_proB_ord<-ordinate(pre_proB_rare_fil, method="PCoA", "wunifrac")

plot_ordination(pre_proB_rare_fil, pre_proB_ord, color="Pre_PostB")

proA_B_rare_fil<-subset_samples(depression2017_fil, ProA_B %in% c("A","B"))
proA_B_ord<-ordinate(proA_B_rare_fil, method="PCoA", "wunifrac")

plot_ordination(proA_B_rare_fil, proA_B_ord, color="ProA_B")

#severity vs moderate vs mild
pre_bdi_levels_rare_fil<-subset_samples(depression2017_fil, PRE_BDI_LEVELS %in% c("severe","moderate", "mild"))
pre_bdi_levels_ord<-ordinate(pre_bdi_levels_rare_fil, method="PCoA", "wunifrac")

plot_ordination(pre_bdi_levels_rare_fil, pre_bdi_levels_ord, color="PRE_BDI_LEVELS")

```


```{r Plotting}

#Transform an OTU table from counts to proportions between 0 and 1.
depression2017_rel_1<-transform_sample_counts(depression2017_fil, function(x) x/sum(x))

depression2017_rel_2<-transform_sample_counts(depression2017_rare, function(x) x/sum(x))


depression2017_rel<-transform_sample_counts(depression2017_1, function(x) x/sum(x))
                                            
#I use the psmelt function to get the data into a form that ggplot likes, dplyr to filter that data I want, and ggplot to do the plotting
#Melt data into a "long” form. I used the OTU table rarefied to 28170 and converted to relative abundance.
depression_smelt<-psmelt(depression2017_rel_1)

depression_smelt2<-psmelt(depression2017_rel_2)

depression_smelt_1<-psmelt(depression2017_rel)
 ##Warning message:
##In psmelt(depression2017_rel_1) : The sample variables: 
##Sample
 ##have been renamed to: 
##sample_Sample
##to avoid conflicts with special phyloseq plot attribute names.

head(depression_smelt)


#Look for Bacillus OTUs in data since it was lost from mock community.
bacillus_otus<-depression_smelt %>% filter(Genus == "g__Bacillus") %>%
  ggplot(aes(x=Sample,y=Abundance))+geom_bar(stat="identity")+facet_grid(.~Participant, scales="free")
  
fp_otus<-depression_smelt %>% filter(Species=="s__prausnitzii")

ggplot(fp_otus, aes(x=Sample,y=Abundance))+geom_bar(stat="identity")+facet_grid(.~Participant, scales="free")

ggplot(fp_otus, aes(x=Participant,y=Abundance))+geom_boxplot(stat = "boxplot")

fp_otus %>% group_by(SampleID) %>% mutate(sum_abund=sum(Abundance)) %>%
  distinct(SampleID, .keep_all=TRUE) %>%
  ggplot(aes(x=Participant,y=Abundance))+geom_boxplot(stat = "boxplot")

f_otus<-depression_smelt %>% filter(Genus=="g__Faecalibacterium")

ggplot(f_otus, aes(x=Species, y=Abundance, fill=Species))+geom_bar(stat="identity")

#Let's plot the overall taxa by group (average relative abundance per group) and colour by phyla
#plot every sample by phyla, faceting into disease groups, 
phyla_plot<-depression_smelt %>% group_by(sample_Sample,Phylum) %>%
  mutate(abund=sum(Abundance)) %>% distinct(Phylum, .keep_all=TRUE) %>% select(sample_Sample, Phylum, Abundance, abund, Participant) %>% arrange(sample_Sample,Phylum)

phyla_plot_rare<-depression_smelt_rare %>% group_by(sample_Sample,Phylum) %>%
  mutate(abund=sum(Abundance)) %>% distinct(Phylum, .keep_all=TRUE) %>% select(sample_Sample, Phylum, Abundance, abund, Participant) %>% arrange(sample_Sample,Phylum)

ggplot(phyla_plot_rare, aes(x=sample_Sample, y=abund, fill=Phylum))+geom_bar(stat="identity") + facet_grid(PostA_B~Participant, scales="free")

probiotics<- depression_smelt2 %>% filter(Genus %in% c("g__Bifidobacterium","g__Lactobacillus", "g__Lacotococcus")) %>% group_by(sample_Sample, Genus) %>% mutate(sum_abund= sum(Abundance)) %>% select(OTU,Sample,Abundance,Participant,ProA_B,Pre_ProA,Pre_PostB,Genus,sum_abund) %>% filter(Participant %in% c("Pre","Post"))
probiotics$Participant<-factor(probiotics$Participant, levels=c("Pre","Post"))
probiotics %>% filter(Pre_ProA%in% c("A","pre")) %>%
  ggplot(aes(x=sample_Sample,y=sum_abund, fill=Genus))+geom_bar(stat="identity")+facet_grid(.~Participant)+labs(title="Group A pre and post", y= "% relative abundance", x="Sample")+
  theme(axis.text.x = element_text(angle=90, size=5))


probiotics_bif<- depression_smelt2 %>% filter(Genus %in% c("g__Bifidobacterium")) %>% group_by(Sample, Genus) %>% mutate(sum_abund= sum(Abundance)) %>% select(OTU,Sample,Abundance,Participant,Participant_Group,ProA_B,Pre_ProA ,Pre_PostB,Genus,sum_abund, Allocation_All, Allocation_Groups) %>% filter(Participant %in% c("Pre","Post")) %>% filter(Pre_ProA!="NA")



ggplot(probiotics_bif, aes(x=Participant_Group, y=Abundance, fill=Allocation_Groups)) + geom_bar(stat="identity", position="dodge")+labs(title="Bifidobacterium Pre and Post Probiotic Treatment", y= "Abundance", x="Sample")+theme(axis.text.x = element_text(angle=90, size=10))

probiotics_bif$Allocation_Groups<-factor(probiotics_bif$Allocation_Groups, levels = c("PreA", "PostA"))

ggplot(probiotics_bif, aes(x=Participant_Group, y=Abundance, fill=factor(Allocation_Groups))) + geom_bar(stat="identity", position="dodge")+labs(title="Bifidobacterium Pre and Post Probiotic Treatment", y= "Abundance", x="Sample", fill="Pre vs Post")+theme(axis.text.x = element_text(angle=90, size=10))




probiotics$Participant<-factor(probiotics$Participant, levels=c("Pre","Post"))


probiotics %>% filter(Pre_ProA%in% c("A","pre")) %>%
  ggplot(aes(x=sample_Sample,y=sum_abund, fill=Genus))+geom_bar(stat="identity")+facet_grid(.~Participant)+labs(title="Group A pre and post", y= "% relative abundance", x="Sample")+ theme(axis.text.x = element_text(angle=90, size=5))

probiotics %>% filter(Pre_PostB%in% c("B","pre")) %>%
  ggplot(aes(x=sample_Sample,y=sum_abund, fill=Genus))+geom_bar(stat="identity")+facet_grid(.~Participant)+labs(title="Group B pre and post", y= "% relative abundance", x="Sample")+
  theme(axis.text.x = element_text(angle=90, size=5))
 
proA<-probiotics %>% filter(Pre_ProA %in% c("A","pre"))

kruskal.test(Abundance ~ Pre_ProA, proA)

proB<-probiotics %>% filter(Pre_PostB %in% c("B","pre"))
kruskal.test(Abundance ~ Pre_PostB, proB)

ggplot(probiotics, aes(x=sample_Sample, y=Abundance, fill=Genus))+geom_bar(stat="identity") + facet_grid(.~ProA_B, scales="free")

ggplot(probiotics, aes(x=ProA_B, y=sum_abund, fill=Genus))+geom_boxplot() 
                                              
roseburia<-depression_smelt_rare %>% filter(OTU=="518438") %>% arrange(desc(Abundance))

depression_smelt_rare_counts<-psmelt(depression2017_rare)

roseburia_counts<-depression_smelt_rare_counts %>% filter(OTU=="518438") %>% arrange(desc(Abundance))

head(roseburia)
ggplot(roseburia_counts, aes(x=Abundance, y=CogReact.Total.Score))+geom_point()

lacno_counts<-depression_smelt_rare_counts %>% filter(OTU=="205148") %>% arrange(desc(Abundance)) %>% filter(Participant!="Con")

ggplot(lacno_counts, aes(x=Abundance, y=CogReact.Total.Score))+geom_point()+geom_smooth(method=lm)



roseburia_counts_DepDass<-depression_smelt_rare_counts %>% filter(OTU=="518438") %>% arrange(desc(Abundance)) %>% filter(Participant!="Post")

head(roseburia_counts_DepDass)
tail(roseburia_counts_DepDass)

ggplot(roseburia_counts_DepDass, aes(x=Abundance, y=DEP_DASS))+geom_point() + labs(title="Dep_Dass Pre vs. Con Rosburia 518438")+geom_smooth(method=lm)

ruminococcus_counts_DepDass<-depression_smelt_rare_counts %>% filter(OTU=="4341886") %>% arrange(desc(Abundance)) %>% filter(Participant!="Post")
ggplot(ruminococcus_counts_DepDass, aes(x=Abundance, y=DEP_DASS))+geom_point() + labs(title="Dep_Dass Pre vs. Con Ruminococcus 4341886")+geom_smooth(method=lm)


clostridiales_counts_cogn<-depression_smelt_rare_counts %>% filter(OTU=="358594") %>% arrange(desc(Abundance)) %>% filter(Participant!="Post")
ggplot(clostridiales_counts_cogn, aes(x=Abundance, y=CogReact.Total.Score))+geom_point() + labs(title="CogReact Pre vs. Con Clostridiales 358594")+geom_smooth(method=lm)

Ruminococcaceae_counts_cogn<-depression_smelt_rare_counts %>% filter(OTU=="295748")

ggplot(Ruminococcaceae_counts_cogn, aes (x=PRE_BDI_LEVELS, y=Abundance))+geom_boxplot()+geom_point()

Top10OTUs<-names(sort(taxa_sums(depression2017_rare),TRUE))[1:10]
top_10<-depression_smelt_rare %>% filter(OTU %in% Top10OTUs)
ggplot(top_10, aes(x=OTU, y=Abundance, fill=Participant))+geom_bar(stat="identity", position="dodge")

Top100OTUs<-names(sort(taxa_sums(depression2017_rare),TRUE))[1:100]
top_100<-depression_smelt_rare_counts %>% filter(OTU %in% Top100OTUs)

ggplot(top_10, aes(x=OTU, y=Abundance, fill=Participant))+geom_boxplot()

```


```{r Writing to a textfile}
write.table((depression_smelt_1 %>% distinct(OTU)), file="all_otus1.txt", sep="\t")
```

```{r}

Top10OTUs<-names(sort(taxa_sums(depression2017_rare),TRUE))[1:10]

top_100<-depression_smelt_rare %>% filter(OTU %in% Top10OTUs)

top_100_PreCon<-top_100 %>% filter(Participant!="Post")

top_100_gnavus<-top_100_PreCon %>% filter(OTU=="360015")

ggplot(top_100_gnavus, aes(x=OTU, y=Abundance, fill=PRE_BDI_LEVELS)) %>% +geom_bar(stat="identity", position="dodge")

ggplot(top_100_gnavus, aes(x=OTU, y=Abundance, fill=PRE_BDI_LEVELS)) %>% +geom_boxplot()


ggplot(top_100_gnavus, aes(x=Abundance, y=DEP_DASS, label=sample_Sample))+geom_point(aes(colour=PRE_BDI_LEVELS)) + labs(title="DEP_DASS correlation with R. gnavus 360015")+geom_smooth(method=lm)


depression_smelt_rare_counts %>% mutate(meanAbund= mean(Abundance)) %>% filter(Family=="f__Ruminococcaceae") %>% filter(PRE_BDI_LEVELS!=("NA")) %>% filter(PRE_BDI_LEVELS!=("moderate")) %>%ggplot(aes(x=PRE_BDI_LEVELS, y=meanAbund, fill=PRE_BDI_LEVELS))+ geom_boxplot()+labs(title="Mean Abundance of f_Ruminococcaceae Severe & Mild BDI")

depression_smelt_rare_counts %>% mutate(meanAbund= mean(Abundance)) %>% filter(Family=="f__Ruminococcaceae") %>% filter(PRE_BDI_LEVELS!=("NA")) %>% filter(PRE_BDI_LEVELS!=("moderate")) %>%ggplot(aes(x=PRE_BDI_LEVELS, y=Abundance, fill=PRE_BDI_LEVELS))+ geom_point()+labs(title="Mean Abundance of f_Ruminococcaceae Severe & Mild BDI")


```

